netplan/cli/sriov.py

```diff
@@ -230,11 +400,17 @@ def apply_vlan_filter_for_vf(pf, vf, vlan_name, vlan_id, prefix='/'):
             'failed setting SR-IOV VLAN filter for vlan %s (ip link set command failed)' % vlan_name)
 
 
-def apply_sriov_config(config_manager):
+def apply_sriov_config(config_manager, rootdir='/'):
     """
     Go through all interfaces, identify which ones are SR-IOV VFs, create
     them and perform all other necessary setup.
     """
+    parser = libnetplan.Parser()
+    parser.load_yaml_hierarchy(rootdir)
+
+    np_state = libnetplan.State()
+    np_state.import_parser_results(parser)
+
```

Comment:
thought: potential regressions here (same as #255, really) with newer, stricter checks

Reply:
ACK. That was done conscientiously. I had all of the new SR-IOV functionality implemented in a way not using libnetplan parser, but felt it would be better if new functionality would be using the new parser (so no rework is needed later on). A potential regression is limited to the SR-IOV usecase and I could relatively easily change that to using just native python `configmanager.py` in case we hit any issues.

We're also exposing the same potential regression for the new `netplan get` usecase/CLI, which could be mitigated by reverting 331ca01908b68446a9a1f395616136678e617a1a in case of any issue.

But I actually like the idea of enabling the new parser in _some_ selected usecases, to give it some limited exposure before enabling it throughout the stack.

Reply:
ACK, I'm just happy I'm not the one introducing potential regressions for once :smile: